From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tomi Valkeinen <tomi.valkeinen@ti.com>
Date: Mon, 31 May 2021 16:31:35 +0530
Subject: [PATCH] arm64: dts: ti: k3-am65-main: fix DSS irq trigger type

DSS irq trigger type is set to IRQ_TYPE_EDGE_RISING. For some reason this
results in double the amount of expected interrupts, e.g. for normal
page flipping test the number of interrupts per second is 2 * fps. It is
as if the IRQ would trigger on both edges.

In any case, it's better to use IRQ_TYPE_LEVEL_HIGH, like the other
devices do, which seems to fix this problem.

Signed-off-by: Tomi Valkeinen <tomi.valkeinen@ti.com>
Signed-off-by: Jyri Sarha <jsarha@ti.com>
Signed-off-by: Nikhil Devshatwar <nikhil.nd@ti.com>
Signed-off-by: Vignesh Raghavendra <vigneshr@ti.com>
Tested-by: Praneeth Bajjuri <praneeth@ti.com>
---
 arch/arm64/boot/dts/ti/k3-am65-main.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/ti/k3-am65-main.dtsi b/arch/arm64/boot/dts/ti/k3-am65-main.dtsi
index fb94d61eae0e..9f8031fe6d08 100644
--- a/arch/arm64/boot/dts/ti/k3-am65-main.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-am65-main.dtsi
@@ -860,7 +860,7 @@ dss: dss@4a00000 {
 		assigned-clocks = <&k3_clks 67 2>;
 		assigned-clock-parents = <&k3_clks 67 5>;
 
-		interrupts = <GIC_SPI 166 IRQ_TYPE_EDGE_RISING>;
+		interrupts = <GIC_SPI 166 IRQ_TYPE_LEVEL_HIGH>;
 
 		dma-coherent;
 
